plugins/modules/foreman_scc_account.py

```diff
@@ -0,0 +1,173 @@
+    default: present
+    choices: ["present", "absent", "synced"]
+    type: str
+extends_documentation_fragment:
+  - foreman
+'''
+
+EXAMPLES = '''
+- name: "Create an scc_account"
+  foreman_scc_account:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "Test"
+    login: "abcde"
+    scc_account_password: "12345"
+    base_url: "https://scc.suse.com"
+    state: present
+
+- name: "Update a scc_account"
+  foreman_scc_account:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "Test1"
+    state: present
+
+- name: "Delete a scc_account"
+  foreman_scc_account:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "Test"
+    state: absent
+'''
+
+RETURN = ''' # '''
+
+
+from ansible.module_utils.foreman_helper import KatelloEntityAnsibleModule
+
+
+def main():
+    module = KatelloEntityAnsibleModule(
+        entity_spec=dict(
+            name=dict(required=True),
+            updated_name=dict(),
+            login=dict(),
+            scc_account_password=dict(no_log=True, flat_name='password'),
+            base_url=dict(),
+            sync_date=dict(),
+            interval=dict(choices=['never', 'daily', 'weekly', 'monthly']),
+            state=dict(default='present', choices=['present', 'absent', 'synced']),
+        ),
+        argument_spec=dict(
+            test_connection=dict(type='bool', default=False)),
+    )
+
+    entity_dict = module.clean_params()
+
+    module.connect()
+
+    entity_dict['organization'] = module.find_resource_by_name('organizations', entity_dict['organization'], thin=True)
+
+    scope = {'organization_id': entity_dict['organization']['id']}
+
+    entity = module.find_resource_by_name('scc_accounts', name=entity_dict['name'], params=scope, failsafe=True)
+
+    if entity_dict['test_connection'] and not module.desired_absent:
+        if not entity:
+            scc_account_credentials = {'login': entity_dict['login'], 'password': entity_dict['scc_account_password'], 'base_url': entity_dict['base_url']}
+            module.resource_action('scc_accounts', 'test_connection', scc_account_credentials, ignore_check_mode=True)
+        else:
+            scc_account_credentials = {'id': entity['id']}
+            if 'login' in entity_dict:
+                scc_account_credentials['login'] = entity_dict['login']
+            if 'scc_account_password' in entity_dict:
+                scc_account_credentials['password'] = entity_dict['scc_account_password']
+            module.resource_action('scc_accounts', 'test_connection', scc_account_credentials, ignore_check_mode=True)
+
+    if 'updated_name' in entity_dict:
+        entity_dict['name'] = entity_dict['updated_name']
+
+    if module.state == 'synced':
+        task = module.resource_action('scc_accounts', 'sync', {'id': entity['id']})
+        if task['result'] == 'error':
```

Comment:
I *think* this should check for `task['result'] != 'success'` instead, to also catch `result=warning` etc.

see also https://github.com/theforeman/foreman-tasks/blob/904c07ca7f2be446e5453eeb271f9c4bc98628de/app/models/foreman_tasks/task/status_explicator.rb#L4-L8 for an alternative "definition" of "errored task"

Reply:
oh, and back in nailgun, we did https://github.com/SatelliteQE/nailgun/blob/master/nailgun/entity_mixins.py#L111-L116